synapse/storage/schema/main/delta/59/14refresh_tokens.sql.postgres

```diff
@@ -0,0 +1,18 @@
+/* Copyright 2021 The Matrix.org Foundation C.I.C
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+-- Add a flag whether the token was already used or not
+ALTER TABLE "access_tokens"
+  ADD COLUMN used BOOLEAN NOT NULL DEFAULT FALSE;
```

Comment:
since github is obstinately refusing to make my reply visible, I'll write it again:



>  Seems like we already did a lot of ADD COLUMN ... NOT NULL DEFAULT ... in the past:

Many of those are small tables, so can be rewritten during Synapse restart without major problems. I'm surprised to see a couple of quite large tables in that list, and I wonder if they caused us trouble at the time, but I don't think that changes my position on it here: I'd like us to avoid rewriting the access_tokens table if possible.

>  Looking at the existing columns with defaults, seems like we don't have that many nullable columns with defaults:

To be clear, to avoid having to rewrite the table, you need to make it nullable with no default.

Reply:
done in https://github.com/matrix-org/synapse/pull/9450/commits/18628fc3adeb28fb820c4e5e1da3b8bece4ce40f